sktime/tests/test_all_estimators.py

```diff
@@ -348,9 +348,14 @@ def test_fit_returns_self(estimator_instance):
 def test_raises_not_fitted_error(estimator_instance):
     """Check that we raise appropriate error for unfitted estimators."""
     estimator = estimator_instance
+
     # call methods without prior fitting and check that they raise our
     # NotFittedError
     for method in NON_STATE_CHANGING_METHODS:
+        # if has fit-in-transform tag (empty fit), no error needs to be raised
+        if method == "transform" and estimator.get_tag("fit-in-transform"):
```

Comment:
I think we still want the workflow to first call `fit` and then `transform`, even if `fit` is empty. Changing that seems like a big risk.

Reply:
can you give me then the definition of the tag `"fit-in-transform"`? Please give a precise spec. What does the tag tell us about a transformer?

Reply:
I'm not sure, perhaps we have to come up with a precise definition. I think it tries to capture the fact that in some transformers, `transform` will change parameters, something that usually is only happening in `fit`.

Reply:
That's not what I thought - I thought it just meant that `fit` was empty.
Who came up with it? Was it you, @aiwalter?

Reply:
I would be in favour of all methods requiring either a call to `fit` or `fit_transform`.
But is this your understanding, i.e., `transform` should *always* raise an error, even if `fit` is empty?

Reply:
I think it was me but doesn't really matter who came up with it. So I don't think we should move aways from always requiring fit to be called first, that would be a big interface change.

Reply:
fine with me - I thought you wanted it to work it this way.
I'll just switch it out to "`fit` must *always* be called even if empty". Would be great if you could confirm if that's what you want.

Reply:
changed it, if it's right now in your opinion, please resolve

Reply:
yes that's fine, can't really comment on the other changes

Reply:
?